### PR TITLE
[[ Datagrid ]] Add docs for datagrid to IDE API Dictionary

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

**Comment by montegoulding** [comment]
Has Paul signed the contributor agreement? Also I presume the parser doesn't mind the extra newline after `Description:`

**Comment by livecodeali** [comment]
Hmm, good point. I've asked him on the list. The parser does indeed not mind the extra newline!

File: Documentation/dictionary/datagrid.lcdoc [hunk 1]
```diff
@@ -0,0 +1,2592 @@
...
+Summary: Set to true to allow users to edit the cells of a table or the
+fields in a form.
+
+Description:
+
+Set to true to allow users to edit the cells of a table or the fields in
+a form. Note that when customizing templates for form rows or table
+columns you would check this property to determine if fields should
+allow editing. The default Row Behavior script contains an Example of
+
```
Comment (peter-b) [inline Documentation/dictionary/datagrid.lcdoc]: Extra blank line.

**Comment by peter-b** [comment]
@livecode-vulcan review ok ba66472f3bd61345860869749667f38d27c18c35

There are some obvious problems (like lots of missing cross-references, and probably some of the formatting could be better), but having this documentation present is a massive step forward and the issues can easily be addressed later.

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok ba66472f3bd61345860869749667f38d27c18c35

**Pull request merged**